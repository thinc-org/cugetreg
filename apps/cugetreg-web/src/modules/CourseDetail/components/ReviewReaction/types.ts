import { ReviewInteractionType } from '@web/common/types/reviews'

export interface ReviewReactionProps {
  type: ReviewInteractionType
  pressed: boolean
  reactionCount: number
  onClick?: () => void
}
