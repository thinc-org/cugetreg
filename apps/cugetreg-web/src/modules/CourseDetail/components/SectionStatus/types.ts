import { Capacity } from '@cugetreg/codegen'

import { TypographyProps } from '@mui/material'

export type AvailableStatus = 'available' | 'full' | 'closed'
export interface SectionStatusProps extends TypographyProps {
  capacity: Capacity
  closed: boolean
}
