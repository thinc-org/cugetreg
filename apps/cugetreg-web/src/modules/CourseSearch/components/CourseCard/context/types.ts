import { Capacity, Course, DayOfWeek, Section } from '@libs/codegen'

export interface CourseCardContextValue {
  course: Course
  isGenEd: boolean
  classDays: DayOfWeek[]
  courseCapacity: Capacity
  sectionNumbers: string[]
  selectedSectionNumber: string
  setSectionNumber: (sectionNo: string) => void
  selectedSection: Section
  teachers: string[]
  isExpanded: boolean
  onToggle: () => void
}
