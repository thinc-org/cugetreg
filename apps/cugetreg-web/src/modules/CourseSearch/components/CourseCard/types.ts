import { Course } from '@cugetreg/codegen'

export interface CourseCardProps {
  course: Course
  rating?: number
}
