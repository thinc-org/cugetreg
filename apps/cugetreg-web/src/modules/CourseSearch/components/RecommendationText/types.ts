import { Semester, StudyProgram } from '@thinc-org/chula-courses'

export interface SelectedCourse {
  courseNo: string
  semesterKey: {
    semester: Semester
    studyProgram: StudyProgram
    academicYear: string
  }
}
