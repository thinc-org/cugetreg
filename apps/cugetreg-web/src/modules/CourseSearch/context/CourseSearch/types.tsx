import { SearchCourseQueryResult } from '@cugetreg/codegen'

export interface CourseSearchProps {
  fetchMoreCourses: () => void
  courseSearchQuery?: SearchCourseQueryResult
}
