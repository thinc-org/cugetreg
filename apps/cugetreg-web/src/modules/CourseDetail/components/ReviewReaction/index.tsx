import { IconBaseProps } from 'react-icons/lib'
import { MdThumbDown, MdThumbDownOffAlt, MdThumbUp, MdThumbUpOffAlt } from 'react-icons/md'

import { IconButton, Stack, Typography, useTheme } from '@mui/material'

import { ReviewInteractionType } from '@web/common/types/reviews'

import { ReviewReactionProps } from './types'

export const ReviewReaction: React.FC<ReviewReactionProps> = ({
  type,
  pressed,
  reactionCount,
  onClick,
}) => {
  const theme = useTheme()

  const getColor = () => {
    if (pressed) {
      switch (type) {
        case ReviewInteractionType.Like:
          return theme.palette.highlight.green[500]
        case ReviewInteractionType.Dislike:
          return theme.palette.highlight.red[500]
      }
    }
    return theme.palette.primaryRange[50]
  }

  const SelectedIcon = (props: IconBaseProps) => {
    const color = getColor()
    const modifiedProps: IconBaseProps = { ...props, color, size: 20 }
    switch (type) {
      case ReviewInteractionType.Like:
        return !pressed ? <MdThumbUpOffAlt {...modifiedProps} /> : <MdThumbUp {...modifiedProps} />
      case ReviewInteractionType.Dislike:
        return !pressed ? (
          <MdThumbDownOffAlt {...modifiedProps} />
        ) : (
          <MdThumbDown {...modifiedProps} />
        )
      default:
        return !pressed ? <MdThumbUpOffAlt {...modifiedProps} /> : <MdThumbUp {...modifiedProps} />
    }
  }

  return (
    <Stack spacing={0.5} direction="row" alignItems="center" onClick={onClick}>
      <IconButton size="small">
        <SelectedIcon />
      </IconButton>
      <Typography variant="h6" color={getColor()}>
        {reactionCount}
      </Typography>
    </Stack>
  )
}
