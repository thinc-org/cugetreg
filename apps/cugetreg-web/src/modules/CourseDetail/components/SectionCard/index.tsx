import { useTranslation } from 'react-i18next'

import {
  Card,
  CardActions,
  CardContent,
  CardHeader,
  Grid,
  Stack,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  Typography,
  styled,
} from '@mui/material'

import { Course, Section } from '@cugetreg/codegen'

import { Caption } from '@web/common/components/Caption'
import { GenEdChip } from '@web/common/components/Chips/catagories/GenEdChip'
import { SelectButton } from '@web/common/components/SelectButton'
import { UnstyledTable } from '@web/common/components/UnstyledTable'
import { dayOfWeekMapper } from '@web/common/constants/dayOfWeek'
import { Analytics } from '@web/common/context/Analytics/components/Analytics'
import { SUBJECT_SELECT_BUTTON_WITH_SECTION } from '@web/common/context/Analytics/constants'
import { getClassPeriod } from '@web/common/utils/getClassPeriod'

import { SectionStatus } from '../SectionStatus'

const SectionTitle = styled(Typography)`
  margin-right: ${({ theme }) => theme.spacing(2)};
`

interface SectionCardProps {
  section: Section
  course: Course
  className?: string
}

export const SectionCard = (props: SectionCardProps) => {
  const { section, course, className } = props
  const { t } = useTranslation('sectionCard')

  return (
    <Card variant="outlined" className={className}>
      <CardHeader
        sx={{ p: { xs: 2, sm: 4 }, pb: { xs: 0, sm: 0 }, pt: { xs: 2, sm: 3 } }}
        title={
          <Stack direction="row" justifyContent="space-between">
            <Grid container spacing={1}>
              <Grid item>
                <SectionTitle variant="h5">
                  {t('section', { sectionNo: section.sectionNo })}
                </SectionTitle>
              </Grid>
              {section.genEdType !== 'NO' && (
                <Grid item>
                  <GenEdChip type={section.genEdType} />
                </Grid>
              )}
            </Grid>
            <SectionStatus capacity={section.capacity} closed={section.closed} />
          </Stack>
        }
      />
      <CardContent sx={{ px: { xs: 2, sm: 4 }, py: 2, pt: { xs: 2, sm: 1 } }}>
        <Stack direction={['column', 'row']} justifyContent="space-between">
          <UnstyledTable>
            <TableHead>
              <TableRow>
                <TableCell>
                  <Caption>{t('teacher')}</Caption>
                </TableCell>
                <TableCell>
                  <Caption>{t('time')}</Caption>
                </TableCell>
                <TableCell>
                  <Caption>{t('classRoom')}</Caption>
                </TableCell>
                <TableCell>
                  <Caption>{t('classType')}</Caption>
                </TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {section.classes.map((sectionClass, index) => (
                <TableRow key={index}>
                  <TableCell>
                    <Typography variant="body1" sx={{ maxWidth: '15ch' }}>
                      {sectionClass.teachers.join(', ')}
                    </Typography>
                  </TableCell>
                  <TableCell>
                    <Typography variant="body1" key={index}>
                      {sectionClass.dayOfWeek && dayOfWeekMapper[sectionClass.dayOfWeek]}{' '}
                      {getClassPeriod(sectionClass)}
                    </Typography>
                  </TableCell>
                  <TableCell>
                    <Typography variant="body1" key={index}>
                      {sectionClass.building} {sectionClass.room}
                    </Typography>
                  </TableCell>
                  <TableCell>
                    <Typography variant="body1" key={index}>
                      {sectionClass.type}
                    </Typography>
                  </TableCell>
                </TableRow>
              ))}
            </TableBody>
          </UnstyledTable>
          <Stack
            sx={{
              display: { xs: 'none', sm: 'flex' },
              justifyContent: 'flex-end',
              flexBasis: 'auto',
              ml: 2,
            }}
          >
            <Analytics
              elementName={SUBJECT_SELECT_BUTTON_WITH_SECTION}
              pathId={course.courseNo}
              elementId={section.sectionNo}
            >
              {({ log }) => (
                <SelectButton log={log} course={course} selectedSectionNumber={section.sectionNo} />
              )}
            </Analytics>
          </Stack>
        </Stack>
      </CardContent>
      <CardActions sx={{ p: { xs: 2, sm: 4 }, pt: { xs: 0, sm: 0 }, display: { sm: 'none' } }}>
        <Analytics
          elementName={SUBJECT_SELECT_BUTTON_WITH_SECTION}
          pathId={course.courseNo}
          elementId={section.sectionNo}
        >
          {({ log }) => (
            <SelectButton log={log} course={course} selectedSectionNumber={section.sectionNo} />
          )}
        </Analytics>
      </CardActions>
    </Card>
  )
}
