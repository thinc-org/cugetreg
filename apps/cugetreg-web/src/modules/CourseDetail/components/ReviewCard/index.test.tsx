import { MdDelete, MdEdit, MdFlag } from 'react-icons/md'

import { ThemeProvider } from '@mui/material'
import { shallow } from 'enzyme'

import {
  mockMyPendingReviews,
  mockMyRejectedReviews,
  mockMyReviews,
  mockReviews,
} from '@web/__mock__/review'
import { ReviewInteractionType } from '@web/common/types/reviews'
import { lightTheme } from '@web/configs/theme'
import { ReviewCardProps } from '@web/modules/CourseDetail/components/ReviewCard/types'
import { ReviewReaction } from '@web/modules/CourseDetail/components/ReviewReaction'

describe('ReviewCard', () => {
  const mockReviewData = mockReviews[0]
  const mockMyReviewData = mockMyReviews[0]
  const mockMyPendingReviewData = mockMyPendingReviews[0]
  const mockMyRejectedReviewData = mockMyRejectedReviews[0]

  const useThemeSpy = jest.fn(() => lightTheme)
  const setInteractionSpy = jest.fn()
  const reportReviewSpy = jest.fn()
  const deleteMyReviewSpy = jest.fn()
  const editMyReviewSpy = jest.fn()
  const useReviewContextSpy = jest.fn(() => ({
    setInteraction: setInteractionSpy,
    reportReview: reportReviewSpy,
    deleteMyReview: deleteMyReviewSpy,
    editMyReview: editMyReviewSpy,
  }))

  jest.doMock('@web/modules/CourseDetail/context/Review', () => ({
    useReviewContext: useReviewContextSpy,
  }))

  jest.doMock('@mui/material', () => ({
    ...jest.requireActual('@mui/material'),
    useTheme: useThemeSpy,
  }))

  afterEach(() => {
    jest.clearAllMocks()
  })

  async function customShallow(props: ReviewCardProps) {
    const { ReviewCard } = await import('.')
    return shallow(<ReviewCard {...props} />, {
      wrappingComponent: ThemeProvider,
      wrappingComponentProps: { theme: lightTheme },
    })
  }

  it('should render correctly if this is not your review', async () => {
    const wrapper = await customShallow(mockReviewData)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render correctly if this is your approved review', async () => {
    const wrapper = await customShallow(mockMyReviewData)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render correctly if this is your pending review', async () => {
    const wrapper = await customShallow(mockMyPendingReviewData)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render correctly if this is your rejected review', async () => {
    const wrapper = await customShallow(mockMyRejectedReviewData)
    expect(wrapper.text().includes('กรุณาแก้ไข ก่อนส่งรีวิวรายวิชาอีกครั้ง')).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })

  it('should not have edit icon and delete icon but report icon if this is not your review', async () => {
    const wrapper = await customShallow(mockReviewData)
    expect(wrapper.find(MdDelete)).toHaveLength(0)
    expect(wrapper.find(MdEdit)).toHaveLength(0)
    // expect(wrapper.find(MdFlag)).toHaveLength(1)
  })

  it('should not have report icon but edit icon and delete icon but have report icon if this is your review', async () => {
    const wrapper = await customShallow(mockMyReviewData)
    expect(wrapper.find(MdDelete)).toHaveLength(1)
    expect(wrapper.find(MdEdit)).toHaveLength(1)
    expect(wrapper.find(MdFlag)).toHaveLength(0)
  })

  it('should not have rejected message if this is not a rejected review', async () => {
    let wrapper = await customShallow(mockReviewData)
    expect(wrapper.text().includes('กรุณาแก้ไข ก่อนส่งรีวิวรายวิชาอีกครั้ง')).toBeFalsy()
    wrapper = await customShallow(mockMyReviewData)
    expect(wrapper.text().includes('กรุณาแก้ไข ก่อนส่งรีวิวรายวิชาอีกครั้ง')).toBeFalsy()
    wrapper = await customShallow(mockMyPendingReviewData)
    expect(wrapper.text().includes('กรุณาแก้ไข ก่อนส่งรีวิวรายวิชาอีกครั้ง')).toBeFalsy()
  })

  it('should call setInteraction with like interaction if we click like', async () => {
    const wrapper = await customShallow(mockReviewData)
    const likeButton = wrapper.find(ReviewReaction).at(0)
    likeButton.simulate('click')
    expect(setInteractionSpy).toHaveBeenCalledTimes(1)
    expect(setInteractionSpy).toHaveBeenCalledWith(mockReviewData._id, ReviewInteractionType.Like)
  })

  it('should call setInteraction with dislike interaction if we click dislike', async () => {
    const wrapper = await customShallow(mockReviewData)
    const dislikeButton = wrapper.find(ReviewReaction).at(1)
    dislikeButton.simulate('click')
    expect(setInteractionSpy).toHaveBeenCalledTimes(1)
    expect(setInteractionSpy).toHaveBeenCalledWith(
      mockReviewData._id,
      ReviewInteractionType.Dislike
    )
  })

  it('should call deleteMyReview when click delete icon', async () => {
    const wrapper = await customShallow(mockMyReviewData)
    const deleteIcon = wrapper.find(MdDelete).at(0).parent()
    deleteIcon.simulate('click')
    expect(deleteMyReviewSpy).toHaveBeenCalledTimes(1)
    expect(deleteMyReviewSpy).toHaveBeenCalledWith(mockMyReviewData._id)
  })

  // it('should call reportReview when click report icon', async () => {
  //   const wrapper = await customShallow(mockMyReviewData)
  //   const reportIcon = wrapper.find(MdDelete).at(0).parent()
  //   reportIcon.simulate('click')
  //   expect(deleteMyReviewSpy).toHaveBeenCalledTimes(1)
  //   expect(deleteMyReviewSpy).toHaveBeenCalledWith(mockMyReviewData._id)
  // })

  it('should call editReview when click edit icon', async () => {
    const wrapper = await customShallow(mockMyReviewData)
    const editIcon = wrapper.find(MdEdit).at(0).parent()
    editIcon.simulate('click')
    expect(editMyReviewSpy).toHaveBeenCalledTimes(1)
    expect(editMyReviewSpy).toHaveBeenCalledWith(mockMyReviewData._id)
  })
})
