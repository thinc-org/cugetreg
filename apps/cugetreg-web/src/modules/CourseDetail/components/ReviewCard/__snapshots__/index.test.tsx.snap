// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewCard should render correctly if this is not your review 1`] = `
<Styled(ForwardRef(Stack))
  direction="column"
  pending={false}
  spacing={2}
>
  <ForwardRef(Stack)
    direction="row"
    flexWrap="wrap"
    gap={1}
    justifyContent="space-between"
  >
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      spacing={2}
    >
      <Styled(Styled(ForwardRef(Typography)))
        variant="h5"
      >
        2564 ภาคปลาย
      </Styled(Styled(ForwardRef(Typography)))>
    </ForwardRef(Stack)>
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      spacing={1}
    >
      <MdOutlineStar
        size={20}
      />
      <Styled(Styled(ForwardRef(Typography)))
        variant="h5"
      >
        4
      </Styled(Styled(ForwardRef(Typography)))>
      <Styled(ForwardRef(Typography))
        variant="subtitle2"
      >
        จาก 5
      </Styled(ForwardRef(Typography))>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
  <Styled(div)
    dangerouslySetInnerHTML={
      Object {
        "__html": "เหนื่อยากเลยค่าแงงงงง",
      }
    }
  />
  <ForwardRef(Stack)
    alignItems="center"
    direction="row"
    justifyContent="space-between"
  >
    <ForwardRef(Stack)
      direction="row"
      spacing={3}
    >
      <ReviewReaction
        onClick={[Function]}
        pressed={false}
        reactionCount={0}
        type="L"
      />
      <ReviewReaction
        onClick={[Function]}
        pressed={false}
        reactionCount={0}
        type="D"
      />
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
</Styled(ForwardRef(Stack))>
`;

exports[`ReviewCard should render correctly if this is your approved review 1`] = `
<Styled(ForwardRef(Stack))
  direction="column"
  pending={false}
  spacing={2}
>
  <ForwardRef(Stack)
    direction="row"
    flexWrap="wrap"
    gap={1}
    justifyContent="space-between"
  >
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      spacing={2}
    >
      <Styled(Styled(ForwardRef(Typography)))
        variant="h5"
      >
        2564 ภาคปลาย
      </Styled(Styled(ForwardRef(Typography)))>
    </ForwardRef(Stack)>
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      spacing={1}
    >
      <MdOutlineStar
        size={20}
      />
      <Styled(Styled(ForwardRef(Typography)))
        variant="h5"
      >
        4
      </Styled(Styled(ForwardRef(Typography)))>
      <Styled(ForwardRef(Typography))
        variant="subtitle2"
      >
        จาก 5
      </Styled(ForwardRef(Typography))>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
  <Styled(div)
    dangerouslySetInnerHTML={
      Object {
        "__html": "เหนื่อยากเลยค่าแงงงงง",
      }
    }
  />
  <ForwardRef(Stack)
    alignItems="center"
    direction="row"
    justifyContent="space-between"
  >
    <ForwardRef(Stack)
      direction="row"
      spacing={3}
    >
      <ReviewReaction
        onClick={[Function]}
        pressed={false}
        reactionCount={0}
        type="L"
      />
      <ReviewReaction
        onClick={[Function]}
        pressed={false}
        reactionCount={0}
        type="D"
      />
    </ForwardRef(Stack)>
    <ForwardRef(Stack)
      direction="row"
      ml="auto"
      spacing={1}
    >
      <ForwardRef(IconButton)
        disabled={true}
        onClick={[Function]}
        size="small"
      >
        <MdEdit
          color="#C9CCE4"
          size={24}
        />
      </ForwardRef(IconButton)>
      <ForwardRef(IconButton)
        onClick={[Function]}
        size="small"
      >
        <MdDelete
          color="#C9CCE4"
          size={24}
        />
      </ForwardRef(IconButton)>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
</Styled(ForwardRef(Stack))>
`;

exports[`ReviewCard should render correctly if this is your pending review 1`] = `
<Styled(ForwardRef(Stack))
  direction="column"
  pending={true}
  spacing={2}
>
  <ForwardRef(Stack)
    direction="row"
    flexWrap="wrap"
    gap={1}
    justifyContent="space-between"
  >
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      spacing={2}
    >
      <Styled(Styled(ForwardRef(Typography)))
        variant="h5"
      >
        2564 ภาคปลาย
      </Styled(Styled(ForwardRef(Typography)))>
      <GeneralChip
        type="reviewPending"
      />
    </ForwardRef(Stack)>
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      spacing={1}
    >
      <MdOutlineStar
        size={20}
      />
      <Styled(Styled(ForwardRef(Typography)))
        variant="h5"
      >
        1.5
      </Styled(Styled(ForwardRef(Typography)))>
      <Styled(ForwardRef(Typography))
        variant="subtitle2"
      >
        จาก 5
      </Styled(ForwardRef(Typography))>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
  <Styled(div)
    dangerouslySetInnerHTML={
      Object {
        "__html": "เขียนดี ๆ แล้ว อิอิ",
      }
    }
  />
  <ForwardRef(Stack)
    alignItems="center"
    direction="row"
    justifyContent="space-between"
  >
    <ForwardRef(Stack)
      direction="row"
      spacing={3}
    >
      <ReviewReaction
        onClick={[Function]}
        pressed={true}
        reactionCount={1}
        type="L"
      />
      <ReviewReaction
        onClick={[Function]}
        pressed={false}
        reactionCount={0}
        type="D"
      />
    </ForwardRef(Stack)>
    <ForwardRef(Stack)
      direction="row"
      ml="auto"
      spacing={1}
    >
      <ForwardRef(IconButton)
        disabled={true}
        onClick={[Function]}
        size="small"
      >
        <MdEdit
          color="#C9CCE4"
          size={24}
        />
      </ForwardRef(IconButton)>
      <ForwardRef(IconButton)
        onClick={[Function]}
        size="small"
      >
        <MdDelete
          color="#C9CCE4"
          size={24}
        />
      </ForwardRef(IconButton)>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
</Styled(ForwardRef(Stack))>
`;

exports[`ReviewCard should render correctly if this is your rejected review 1`] = `
<Styled(ForwardRef(Stack))
  direction="column"
  pending={true}
  spacing={2}
>
  <ForwardRef(Stack)
    direction="row"
    flexWrap="wrap"
    gap={1}
    justifyContent="space-between"
  >
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      spacing={2}
    >
      <Styled(Styled(ForwardRef(Typography)))
        variant="h5"
      >
        2564 ภาคปลาย
      </Styled(Styled(ForwardRef(Typography)))>
      <GeneralChip
        type="reviewRejected"
      />
    </ForwardRef(Stack)>
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      spacing={1}
    >
      <MdOutlineStar
        size={20}
      />
      <Styled(Styled(ForwardRef(Typography)))
        variant="h5"
      >
        1.5
      </Styled(Styled(ForwardRef(Typography)))>
      <Styled(ForwardRef(Typography))
        variant="subtitle2"
      >
        จาก 5
      </Styled(ForwardRef(Typography))>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
  <Styled(div)
    dangerouslySetInnerHTML={
      Object {
        "__html": "เขียนดี ๆ แล้ว อิอิ",
      }
    }
  />
  <ForwardRef(Stack)
    alignItems="center"
    direction="row"
    justifyContent="space-between"
  >
    <Styled(ForwardRef(Typography))
      variant="subtitle2"
    >
      กรุณาแก้ไข ก่อนส่งรีวิวรายวิชาอีกครั้ง
    </Styled(ForwardRef(Typography))>
    <ForwardRef(Stack)
      direction="row"
      ml="auto"
      spacing={1}
    >
      <ForwardRef(IconButton)
        disabled={true}
        onClick={[Function]}
        size="small"
      >
        <MdEdit
          color="#C9CCE4"
          size={24}
        />
      </ForwardRef(IconButton)>
      <ForwardRef(IconButton)
        onClick={[Function]}
        size="small"
      >
        <MdDelete
          color="#C9CCE4"
          size={24}
        />
      </ForwardRef(IconButton)>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
</Styled(ForwardRef(Stack))>
`;
